### ISSUE TITLE
Activate extension with workspaceContains when a matching file is added to the workspace

### ISSUE THREAD

**Issue by roblourens** [Microsoft/vscode#28122]
If I have `"workspaceContains:**/*.md"` as an activationEvent, it would be useful for it to fire when a matching file is added to the workspace, not just when a workspace with matching file is opened.

https://github.com/Microsoft/vscode/issues/27665#issuecomment-306394054

**Comment by 1nVitr0** [comment]
Are there any plans to implement this?

I'd rather not have my extension running permanently to figure out if it can be activated. This would be the far better solution regarding performance.

**Comment by JustinGrote** [comment]
As a workaround, you can lazy-load extensions now, so you can lazy load just the detection logic and active that rest on detection change.

Ideally this is first party handled by vscode however.

**Comment by 1nVitr0** [comment]
I was thrown for a loop by the `now`. You mean, that this has always been possible and can be implemented in any extension, regardless of the state of the vscode API?

**Comment by JustinGrote** [comment]
@1nVitr0 https://code.visualstudio.com/api/references/activation-events#onStartupFinished

`onStartupFinished` can be used to lazy load an extension in a way that minimizes vscode startup performance.

**Comment by 1nVitr0** [comment]
@JustinGrote Right, that makes total sense. If combined with the proper `workspaceContains` selector, that should allow the extension to load immediately if necessary or listen for the files without impacting startup performance. Thanks!
